### PR TITLE
Fixed build errors and optional handling

### PR DESCRIPTION
**Pull request by jessearmand**
Overall, my changes minimised the implicit unwrapping of optional values, which may cause a runtime crash, when there's a possibility that the variable is nil.

These issues are indirectly shown by the latest Swift compiler build errors

### CODE REVIEW

File: Code/Controllers/ViewController.swift [hunk 1]
```diff
@@ -219,7 +226,7 @@ class ViewController: UIViewController, PFLogInViewControllerDelegate, PFSignUpV
                 if error == nil {
                     let identityToken = object as! String
                     self.layerClient.authenticateWithIdentityToken(identityToken) { authenticatedUserID, error in
-                        if (!authenticatedUserID.isEmpty) {
+                        if (authenticatedUserID!.isEmpty) {
```
Comment (kwkhaw) [inline Code/Controllers/ViewController.swift]: Hi @jessearmand, Thanks for the PR! :-)
I think here the logic is inverted? We are supposed to check that the authenticatedUserID is not empty, then it is considered successful?

**Comment by jessearmand** [comment]
Oh right, will fix that

**Comment by jessearmand** [comment]
I have changed it, so it validates the failed cases before it completes with the successful one.

**Comment by kwkhaw** [comment]
@jessearmand Great! Thanks for the fix! :-)

**Pull request merged**